sencha-workspace/packages/slate-cbl/src/view/student/Dashboard.js

```diff
@@ -0,0 +1,154 @@
+        '<div class="cbl-progress-percent">{[percent]}%</div>',
+        '<div class="cbl-progress-average" title="Average">{[fm.number(studentCompletion.demonstrationsAverage, "0.##")]}</div>',
+        '</div>',
+
+        '<div class="explainer">',
+        '<p>{Statement}</p>',
+        '</div>',
+
+        '<ul class="cbl-skill-meter skills-unloaded"></ul>',
+        '</div>',
+        '</li>',
+        '</tpl>'
+    ],
+
+    skillsTpl: [
+        '<tpl for=".">',
+        '<li class="cbl-skill">',
+        '<h5 class="cbl-skill-name" data-descriptor="{Descriptor}" data-statement="{Statement}">{Descriptor}</h5>',
+        '<ul class="cbl-skill-demos">',
+        '<tpl for="this.getDemonstrationBlocks(values)">',
+        '<li class="cbl-skill-demo <tpl if="Level==0"> cbl-grid-demo-missed</tpl>" <tpl if="SkillID">data-skill="{SkillID}"</tpl> <tpl if="DemonstrationID">data-demonstration="{DemonstrationID}"</tpl>>',
+        '<tpl if="Level &gt;= 0">',
+        '{[values.Level == 0 ? "M" : values.Level]}',
+        '<tpl else>',
+        '&nbsp;',
+        '</tpl>',
+        '</li>',
+        '</tpl>',
+        '</ul>',
+        '<div class="cbl-skill-description"><p>{Statement}</p></div>',
+//                '<div class="cbl-skill-complete-indicator cbl-level-{parent.level} is-checked">',
+//                    '<svg class="check-mark-image" width="16" height="16">',
+//                        '<polygon class="check-mark" points="13.824,2.043 5.869,9.997 1.975,6.104 0,8.079 5.922,14.001 15.852,4.07"/>',
+//                    '</svg>',
+//                '</div>',
+        '</li>',
+        '</tpl>',
+        {
+            getDemonstrationBlocks: function(skill, studentId) {
+                var demonstrationsRequired = skill.DemonstrationsRequired,
+                    blocks = Ext.isArray(skill.demonstrations) ? skill.demonstrations : [];
+
+                blocks = Ext.Array.sort(blocks, function(a, b) {
+                    return a.Level < b.Level ? -1 : 1;
+                });
+
+                blocks = blocks.splice(0, demonstrationsRequired);
+
+                // sort by most recent, followed by M
+                Ext.Array.sort(blocks, function(a, b) {
+                    if (a.Level == 0) {
+                        return 1;
+                    }
+
+                    if (b.Level == 0) {
+                        return -1;
+                    }
+
+                    // HACK: the ID should not be used in place of a timestamp
+                    return a.DemonstrationID < b.DemonstrationID ? -1 : 1;
+                });
+
+                // add empty blocks
+                while (blocks.length < demonstrationsRequired) {
+                    blocks.push({ID: skill.ID});
+                }
+
+                return blocks;
+            }
+        }
+    ],
+
+    html: [
+        '<ul class="cbl-competency-panels competencies-unloaded" id="studentDashboardCompetenciesList"></ul>'
```

Comment:
@jmealo javascript components should _never_ create DOM elements with static IDs. Ext will assign them on the fly as needed. I would remove the ID and select this via css class.

However we can simplify this a bit and reduce a level of nesting. Ext.Component has a few techniques for generating initial markup and grabbing references to them. Since we only need a reference to the outermost component the best thing to use here is the [`autoEl`](http://docs.sencha.com/extjs/5.0/5.0.0-apidocs/#!/api/Ext.Component-cfg-autoEl) config, which lets you change the DOM element created at the top level for the Ext.Component from its default of a div to any DomHelper spec you want:

```
autoEl: {
    tag: 'ul',
    cls: 'cbl-competency-panels competencies-unloaded'
}
```

Then `this.el` gets you to the Ext.Element you need and no query is necessary